src/error.ts

```diff
@@ -33,13 +34,55 @@ export interface Range {
   readonly length: number;
 }
 
+export interface Location {
+  readonly file: JsonFile;
+  readonly range: Range;
+}
+
+export interface Position {
+  /** 1 indexed */
+  line: number;
+  /** 1 indexed */
+  column: number;
+}
+
+export class OffsetToPositionConverter {
+  readonly newlineIndexes: readonly number[];
```

Comment:
`#private`?

Reply:
This is useful for other stuff too, e.g. could potentially make drawSquiggle more efficient